e2e/project/webhookSessionTriggered.spec.ts

```diff
@@ -0,0 +1,105 @@
+import { APIRequestContext, Page } from "@playwright/test";
+
+import { expect, test } from "../fixtures";
+import { waitForToast } from "../utils";
+import { DashboardPage } from "e2e/pages";
+
+interface SetupParams {
+	dashboardPage: DashboardPage;
+	page: Page;
+	request: APIRequestContext;
+}
+
+async function waitForFirstCompletedSession(page: Page, timeoutMs = 60000) {
+	await expect(async () => {
+		const refreshButton = page.getByRole("button", { name: "Refresh" });
+		const isDisabled = await refreshButton.evaluate((element) => (element as HTMLButtonElement).disabled);
+
+		if (!isDisabled) {
+			await refreshButton.click();
+			await page.waitForTimeout(500);
+		}
+
+		const hasErrorStatus = await page
+			.getByRole("status", { name: "completed" })
+			.filter({ hasText: "1" })
+			.isVisible();
+
+		expect(hasErrorStatus).toBe(true);
+
+		return true;
+	}).toPass({
+		timeout: timeoutMs,
+		intervals: [2000],
+	});
+}
+
+test.describe("Session triggered with webhook", () => {
+	test.beforeEach(async ({ dashboardPage, page, request }: SetupParams) => {
+		await setupProjectAndTriggerSession({ dashboardPage, page, request });
+	});
+
+	test("should successfully deploy project and execute session via webhook", async ({ page }: { page: Page }) => {
+		const deploymentTableRow = page.getByRole("cell", { name: /bld_*/ });
+		await expect(deploymentTableRow).toHaveCount(1);
+		const completedSessionDeploymentColumn = page
+			.getByRole("status", { name: "completed" })
+			.filter({ hasText: "1" });
+		completedSessionDeploymentColumn.click();
+
+		await page
+			.locator("role=row", {
+				has: page.getByRole("cell", { name: "receive_http_get_or_head" }),
+			})
+			.click();
+
+		await page.waitForTimeout(5000);
```

Comment:
why do we need here - `waitForTimeout`?

Reply:
It was added to see the error of the session in the video.
Now when we figured it out, removed the timeout.